### DISCUSSION TITLE
Additional CORS related settings for self-hosted

### DISCUSSION THREAD

**Discussion by mavrick** [outline/outline#1396] [General]
We're using Outline for our internal wiki system but have it hosted behind Cloudflare Access as we have team members across the globe.

For it to work behind Cloudflare Access some CSP connect params need adding such as the `subdomain.cloudflareaccess.com` and changes to the `ApiClient.js` fetch request to send credentials: "same-origin", instead of "omit". See: https://developers.cloudflare.com/access/setting-up-access/cors/#troubleshooting 

I know this might be super edge case. Normally we'd just use an IP restriction however one of our team members is unable to get a static ip address from his ISP and his IP changes daily. I understand the Google login domain can be locked down to a specific `@domain.tld` extension as well, that would solve this however we'd still like it locked down so the subdomain cannot be reached unless authenticated via Cloudflare first.

**Proposed feature request/changes:**

Move CSP/CORS related values to ENV_VARS or allow additional values to be injected just like the `AWS_S3_UPLOAD_BUCKET_URL` and control over the `credentials` variable for fetch requests.

This would also allow people to remove google analytics / sentry access if  they so wished.

**Comment by tommoor** [comment]
Do you have documentation on the CSP params? It doesn't make any sense to me that Cloudflare Access would require changes to the CSP unless they are injecting scripts into the app… If that's the case then almost every application with reasonable security precautions would fail.

**Reply by mavrick** [reply]
I'll have to take a deeper look into why the CSP failed. Off the top of my head I'm assuming the static asset requests are being redirected first before loading causing CSP to fail.
